source/simulator/stokes_matrix_free.cc

```diff
@@ -1606,6 +1674,126 @@ namespace aspect
+          sim.newton_handler->create_material_model_outputs(out);
+#if DEAL_II_VERSION_GTE(9,3,0)
+          const unsigned int n_cells = stokes_matrix.get_matrix_free()->n_cell_batches();
+#else
+          const unsigned int n_cells = stokes_matrix.get_matrix_free()->n_macro_cells();
+#endif
+          const unsigned int n_q_points = quadrature_formula.size();
+
+          active_strain_rate_table.reinit(TableIndices<2>(n_cells, n_q_points));
+          active_viscosity_derivative_wrt_pressure_table.reinit(TableIndices<2>(n_cells, n_q_points));
+          active_viscosity_derivative_wrt_strain_rate_table.reinit(TableIndices<2>(n_cells, n_q_points));
+
+          for (unsigned int cell=0; cell<n_cells; ++cell)
+            {
+#if DEAL_II_VERSION_GTE(9,3,0)
+              const unsigned int n_components_filled = stokes_matrix.get_matrix_free()->n_active_entries_per_cell_batch(cell);
+#else
+              const unsigned int n_components_filled = stokes_matrix.get_matrix_free()->n_components_filled(cell);
+#endif
+              for (unsigned int i=0; i<n_components_filled; ++i)
+                {
+                  typename DoFHandler<dim>::active_cell_iterator FEQ_cell =
+                    stokes_matrix.get_matrix_free()->get_cell_iterator(cell,i);
+                  typename DoFHandler<dim>::active_cell_iterator sim_cell(&(sim.triangulation),
+                                                                          FEQ_cell->level(),
+                                                                          FEQ_cell->index(),
+                                                                          &(sim.dof_handler));
+
+                  fe_values.reinit(sim_cell);
+                  in.reinit(fe_values, sim_cell, sim.introspection, sim.current_linearization_point);
+
+                  sim.material_model->fill_additional_material_model_inputs(in, sim.current_linearization_point, fe_values, sim.introspection);
+                  sim.material_model->evaluate(in, out);
+
+                  const MaterialModel::MaterialModelDerivatives<dim> *derivatives
+                    = out.template get_additional_output<MaterialModel::MaterialModelDerivatives<dim> >();
+
+                  for (unsigned int q=0; q<n_q_points; ++q)
+                    {
+                      const SymmetricTensor<2,dim> strain_rate = in.strain_rate[q];
+
+                      Assert(derivatives != nullptr,
+                             ExcMessage ("Error: The Newton method requires the material to "
+                                         "compute derivatives."));
+
+                      const double viscosity_derivative_wrt_pressure
+                        = derivatives->viscosity_derivative_wrt_pressure[q];
+
+                      const SymmetricTensor<2,dim> viscosity_derivative_wrt_strain_rate
+                        = derivatives->viscosity_derivative_wrt_strain_rate[q];
```

Comment:
I dont think you need these copies or the copy in line 1735 above. Put all of the original names into the lines below and break the lines if they become too long.